### PR TITLE
Fix for Issue 620

### PR DESCRIPTION
**Pull request by ozten**
This PR adds connect-cachify as a dependency.

It simplifies our layout ejs to use a cachify_js and cachify_css helper. The list of assets are managed in lib/static_resources.js.

In production, a single asset file will be served up with /v/<md5 hash>/original/path and expires headers of 1 year.

In development, multiple asset files will be served up with their paths. No special expires headers are sent (just static middleware headers).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +28,7 @@
         "temp": "0.2.0",
         "uglify-js": "1.0.6",
         "uglifycss": "0.0.4",
+        "underscore": "1.3.1",
```
Comment (benadida) [inline package.json:31]: +1 to using underscore more thoroughly everywhere! Much nicer code.

**Comment by benadida** [comment]
this patch looks great to me, especially the way it cleans up the templates.

**Comment by shane-tomlinson** [comment]
+r - backend tests pass, from my understanding of the backend looks good.  

Prod mode requests have md5 hash.
Debug mode requests do not have md5 hash.
External file should enable easy inclusion into a compressor.

**Comment by ozten** [comment]
Thanks! landed on dev.

**Pull request closed without merging**